_pytest/main.py

```diff
@@ -728,17 +730,58 @@ def _tryconvertpyarg(self, x):
         """Convert a dotted module name to path.
 
         """
-        import pkgutil
+        @contextlib.contextmanager
+        def _patched_find_module():
+            """Patch bug in pkgutil.ImpImporter
+
+            When using pkgutil.find_loader on python<3.4 it removes symlinks
+            from the path due to a call to os.path.realpath. This is not consistent
+            with actually doing the import (in these versions, pkgutil and __import__
+            did not share the same underlying code). This can break conftest
+            discovery for pytest where symlinks are involved.
+
+            The only supported python<3.4 by pytest is python 2.7.
+            """
+            if six.PY2:  # python 3.4+ uses importlib instead
+                def find_module_patched(self, fullname, path=None):
+                    subname = fullname.split(".")[-1]
+                    if subname != fullname and self.path is None:
+                        return None
+                    if self.path is None:
+                        path = None
+                    else:
+                        path = [self.path]
+                    try:
+                        file, filename, etc = pkgutil.imp.find_module(subname,
+                                                                      path)
+                    except ImportError:
+                        return None
+                    return pkgutil.ImpLoader(fullname, file, filename, etc)
+
+                old_find_module = pkgutil.ImpImporter.find_module
+                pkgutil.ImpImporter.find_module = find_module_patched
+                try:
+                    yield
+                finally:
+                    pkgutil.ImpImporter.find_module = old_find_module
+            else:
+                yield
+        #
+        # if six.PY2:  # python 3.4+ uses importlib instead
+        #     pkgutil.ImpImporter.find_module = find_module_patched
+
         try:
-            loader = pkgutil.find_loader(x)
+            with _patched_find_module():
+                loader = pkgutil.find_loader(x)
         except ImportError:
             return x
         if loader is None:
             return x
         # This method is sometimes invoked when AssertionRewritingHook, which
         # does not define a get_filename method, is already in place:
         try:
-            path = loader.get_filename(x)
+            with _patched_find_module():
```

Comment:
I'm not 100% sure why it is needed here, but I do know the tests fail if you don't have patching in both places.